reliability/tasks/Apps.py

```diff
@@ -81,19 +84,19 @@ def add(self, app):
                 if not visit_success:
                     self.failed_apps += 1
                     self.logger.error("add_app: " + app.project + "." + app.deployment + " did not become available" )
+        return app
 
     def refresh_stats(self):
+        # need to set 0 because TaskManager will loop this function when pause file exists
+        # the number like self.apps[key].build_count is the total number at any time
+        self.build_count = 0
         for key in self.apps.keys():
             self.build_count += self.apps[key].build_count
-            self.visit_succeded += self.apps[key].visit_succeded
-            self.visit_failed += self.apps[key].visit_failed
 
-    
     # removing an app just removes the dictionary entry, actual app removed by project deletion
     def remove(self,app):
+        # Q: why adding build_count when removing app?
```

Comment:
I think the intention was to make sure any builds that happened since self.build_count was updated were added to the counter. But, it may be a bug.

Reply:
I got it. The total builds in the summary is the total builds ever done during the test, including the builds of the removed apps. Then I think the logic of func refresh_stats is not correct, because it gets all the build numbers of the current existing apps. https://github.com/openshift/svt/blob/fb572b272f3e897eeac1c1b8d45c1dc7fd5ff683/reliability/tasks/Apps.py#L89-L94

I'll modify this counter.

Reply:
Modified build counter in commit https://github.com/openshift/svt/pull/670/commits/1a854b15f85ce36b7b0f45a6fcd42dc511d73e9b